### PR TITLE
feat(condo): DOMA-10237 added maximum payment amount in acquiring integration

### PR DESCRIPTION
**Pull request by vovaaxeapolla**
added a field maximum payment amount in acquiring integration if this field is set then payments greater than its value will not go through

### CODE REVIEW

File: apps/condo/domains/acquiring/schema/AcquiringIntegration.js [hunk 1]
```diff
@@ -79,6 +79,10 @@ const AcquiringIntegration = new GQLListSchema('AcquiringIntegration', {
             ...POSITIVE_MONEY_AMOUNT_FIELD,
             schemaDoc: 'The minimum payment amount that can be accepted',
         },
+        maximumPaymentAmount: {
+            ...POSITIVE_MONEY_AMOUNT_FIELD,
```
Comment (dkoviazin) [inline apps/condo/domains/acquiring/schema/AcquiringIntegration.js:83]: You need to allow null value
Case: We set maximum payment amount, then we want to remove maximum payment amount
You cannot do it through admin ui

File: apps/condo/domains/acquiring/schema/RegisterMultiPaymentForInvoicesService.js [hunk 2]
```diff
@@ -237,13 +241,21 @@ const RegisterMultiPaymentForInvoicesService = new GQLCustomSchema('RegisterMult
...
                     .add(Big(totalAmount.explicitFee))
+
                 if (acquiringIntegration.minimumPaymentAmount && Big(amountToPay).lt(acquiringIntegration.minimumPaymentAmount)) {
                     throw new GQLError({
                         ...ERRORS.PAYMENT_AMOUNT_LESS_THAN_MINIMUM,
                         messageInterpolation: { minimumPaymentAmount: Big(acquiringIntegration.minimumPaymentAmount).toString() },
                     }, context)
                 }
 
+                if (acquiringIntegration.maximumPaymentAmount && Big(amountToPay).gt(acquiringIntegration.maximumPaymentAmount)) {
```
Comment (dkoviazin) [inline apps/condo/domains/acquiring/schema/RegisterMultiPaymentForInvoicesService.js:252]: Why not lte and gte
Minimum payment amount 15.00 - will throw an error that Payment amount must be at least 15.00

File: apps/condo/migrations/20240918174110-0423_acquiringintegration_maximumpaymentamount_and_more.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    `)
+}
+
+exports.down = async (knex) => {
+    await knex.raw(`
+    BEGIN;
+--
+-- Add field maximumPaymentAmount to acquiringintegrationhistoryrecord
+--
+ALTER TABLE "AcquiringIntegrationHistoryRecord" DROP COLUMN "maximumPaymentAmount" CASCADE;
```
Comment (dkoviazin) [inline apps/condo/migrations/20240918174110-0423_acquiringintegration_maximumpaymentamount_and_more.js:36]: why cascade?

File: packages/keystone/fields/SignedDecimal/Implementation.js [hunk 4]
```diff
@@ -16,6 +16,7 @@ class SignedDecimal extends Decimal.implementation {
 
     async validateInput (args) {
         const { resolvedData, fieldPath, addFieldValidationError } = args
+        if (resolvedData[fieldPath] === '') resolvedData[fieldPath] = null
```
Comment (dkoviazin) [inline packages/keystone/fields/SignedDecimal/Implementation.js:19]: Need a comment here, that in another case admin-ui do not allow reset field to null

File: apps/condo/lang/en/en.json [hunk 5]
```diff
@@ -3138,6 +3138,7 @@
   "acquiringReceipt.invoice.row.vatPercent": "VAT",
   "acquiringReceipt.invoice.row.amount": "Amount",
   "api.acquiring.multiPayment.error.receiptsWithInvoices": "Receipts and invoices are forbidden to be together",
-  "api.acquiring.payment.error.paymentAmountLessThanMinimum": "Payment amount must be at least {minimumPaymentAmount}"
+  "api.acquiring.payment.error.paymentAmountLessThanMinimum": "Payment amount must be at least {minimumPaymentAmount}",
+  "api.acquiring.payment.error.paymentAmountGreaterThanMaximum": "Payment amount must be no more than {maximumPaymentAmount}"
```
Comment (dkoviazin) [inline apps/condo/lang/en/en.json:3142]: The payment amount must not exceed {maximumPaymentAmount}

File: apps/condo/migrations/20240918174110-0423_acquiringintegration_maximumpaymentamount_and_more.js (hunk 3 above)
Comment (vovaaxeapolla) [reply apps/condo/migrations/20240918174110-0423_acquiringintegration_maximumpaymentamount_and_more.js:36]: it's autogenerated code

**Review by dkoviazin — COMMENTED** [review]
You need to rebuild the migration

**Review by dkoviazin — APPROVED** [review]

**Review by abshnko — APPROVED** [review]

**Pull request merged**